Taskfile.yml

```diff
@@ -319,10 +319,25 @@ tasks:
   mongosh:
     desc: "Run MongoDB shell (`mongosh`)"
     cmds:
-      - >
-        docker compose exec mongodb mongosh
-        'mongodb://host.docker.internal:27018/test?tls=true&tlsCAFile=/etc/certs/rootCA.pem&heartbeatFrequencyMS=300000'
-        --verbose --eval 'disableTelemetry()' --shell
+      - task: mongosh-plain
+      - task: mongosh-scram
+
+  mongosh-plain:
+    cmds:
+      - cmd: >
+          docker compose exec mongodb mongosh
+          'mongodb://username:password@host.docker.internal:27018/?authMechanism=PLAIN&tls=true&tlsCAFile=/etc/certs/rootCA.pem&heartbeatFrequencyMS=300000'
+          --verbose --eval 'disableTelemetry()' --shell /mongosh/test.js
+        ignore_error: true
+
+  mongosh-scram:
+    cmds:
+      - cmd: >
+          docker compose exec mongodb mongosh
+          'mongodb://username:password@host.docker.internal:27018/?tls=true&tlsCAFile=/etc/certs/rootCA.pem&heartbeatFrequencyMS=300000'
+          --verbose --eval 'disableTelemetry()' --shell /mongosh/test.js
```

Comment:
Perhaps not plain isn't needed for this issue but this seems to return error for me.

```
$ task mongosh-scram
task: [mongosh-scram] docker compose exec mongodb mongosh 'mongodb://username:password@host.docker.internal:27018/?tls=true&tlsCAFile=/etc/certs/rootCA.pem&heartbeatFrequencyMS=300000' --verbose --eval 'disableTelemetry()' --shell /mongosh/test.js

Current Mongosh Log ID: 63a514be1ed5c33944d4e18d
Connecting to:          mongodb://<credentials>@host.docker.internal:27018/?tls=true&tlsCAFile=%2Fetc%2Fcerts%2FrootCA.pem&heartbeatFrequencyMS=300000&directConnection=true&appName=mongosh+1.6.1
/usr/bin/mongosh:98954
  const dict = parsePayload(payload.value());
                                    ^

TypeError: Cannot read properties of undefined (reading 'value')
    at continueScramConversation (/usr/bin/mongosh:98954:37)
    at /usr/bin/mongosh:98921:5
    at Connection.onMessage (/usr/bin/mongosh:98093:5)
    at MessageStream.<anonymous> (/usr/bin/mongosh:97873:56)
    at MessageStream.emit (node:events:513:28)
    at MessageStream.emit (node:domain:489:12)
    at processIncomingData (/usr/bin/mongosh:97686:12)
    at MessageStream._write (/usr/bin/mongosh:97568:5)
    at writeOrBuffer (node:internal/streams/writable:391:12)
    at _write (node:internal/streams/writable:332:10)
```

Reply:
`task mongosh` runs both tasks and ignores errors. `mongosh-plain` works with FerretDB and PLAIN authentication, while `mongosh-scram` works with MongoDB. Both are undocumented because they are implementation details.

I guess we could add comments to explain that

Reply:
Thanks, I connected to FerretDB with `mongosh-scram` so that was what I did wrong :see_no_evil:

Reply:
I've added some comments to the `mongosh` task.